lib/kafka_ex/network_client.ex

```diff
@@ -57,4 +59,15 @@ defmodule KafkaEx.NetworkClient do
       _ -> to_char_list(host)
     end
   end
+
+  defp default_socket_options do
+    default_options = [:binary, {:packet, 4}]
+    is_ssl = Application.get_env(:kafka_ex, :enable_ssl, false)
+    ssl_options = Application.get_env(:kafka_ex, :ssl_options, [])
+    if is_ssl do
+      default_options ++ ssl_options ++ [:ssl]
+    else
+      default_options
+    end
+  end
```

Comment:
hey @comogo,

First of all, I'm a beginner developer in Elixir. 
After that, what about?

```ex
  defp default_socket_options do
    Application.get_env(:kafka_ex, :enable_ssl, false)
    |> default_socket_options_when_ssl_is
  end

  defp default_socket_options_when_ssl_is(true) do
    default_socket_options(false) ++ Application.get_env(:kafka_ex, :ssl_options, []) ++ [:ssl]
  end
  defp default_socket_options_when_ssl_is(false), do: [:binary, {:packet, 4}]
```